activiti-cloud-services-runtime-bundle/activiti-cloud-services-connectors/src/main/java/org/activiti/services/connectors/model/IntegrationEvent.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2017 Alfresco, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.activiti.services.connectors.model;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class IntegrationEvent {
+
+    private String processInstanceId;
+
+    private String processDefinitionId;
+
+    private String correlationId;
+
+    private Map<String, Object> variables;
+
+    public String getProcessInstanceId() {
+        return processInstanceId;
+    }
+
+    public void setProcessInstanceId(String processInstanceId) {
+        this.processInstanceId = processInstanceId;
+    }
+
+    public String getProcessDefinitionId() {
+        return processDefinitionId;
+    }
+
+    public void setProcessDefinitionId(String processDefinitionId) {
+        this.processDefinitionId = processDefinitionId;
+    }
+
+    public String getCorrelationId() {
+        return correlationId;
+    }
+
+    public void setCorrelationId(String correlationId) {
```

Comment:
@erdemedeiros for Events we should avoid setters at all cost. and also.. should we start using Lombok?

Reply:
Yeah, I initially created setters to avoid having ugly constructors with tons of parameters. Now that we are going to remove the correlationId it should be possible to switch to the constructor approach.
Lombok could be interesting, but before using it, I just want to be sure that it does not kill the "call hierarchy" from the IDE.

Reply:
It seems ok: https://stackoverflow.com/questions/42644923/eclipse-with-lombok-search-for-getter-and-setter-usages.

Reply:
👍